Makefile

```diff
@@ -329,28 +345,38 @@ push-latest:
 	done
 
 
+# promtail
 promtail-image:
-	$(SUDO) docker build -t $(IMAGE_PREFIX)/promtail -f cmd/promtail/Dockerfile .
-	$(SUDO) docker tag $(IMAGE_PREFIX)/promtail $(IMAGE_PREFIX)/promtail:$(IMAGE_TAG)
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/promtail:$(IMAGE_TAG) -f cmd/promtail/Dockerfile .
+
+promtail-debug-image: OCI_PLATFORMS=
 promtail-debug-image:
-	$(SUDO) docker build -t $(IMAGE_PREFIX)/promtail -f cmd/promtail/Dockerfile.debug .
-	$(SUDO) docker tag $(IMAGE_PREFIX)/promtail-debug $(IMAGE_PREFIX)/promtail-debug:$(IMAGE_TAG)
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/promtail:$(IMAGE_TAG)-debug -f cmd/promtail/Dockerfile.debug .
+
+promtail-push:
+	$(SUDO) $(PUSH_OCI) $(IMAGE_PREFIX)/promtail:$(IMAGE_TAG)
 
+# loki
 loki-image:
-	$(SUDO) docker build -t $(IMAGE_PREFIX)/loki -f cmd/loki/Dockerfile .
-	$(SUDO) docker tag $(IMAGE_PREFIX)/loki $(IMAGE_PREFIX)/loki:$(IMAGE_TAG)
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/loki:$(IMAGE_TAG) -f cmd/loki/Dockerfile .
+
+loki-debug-image: OCI_PLATFORMS=
 loki-debug-image:
-	$(SUDO) docker build -t $(IMAGE_PREFIX)/loki -f cmd/loki/Dockerfile.debug .
-	$(SUDO) docker tag $(IMAGE_PREFIX)/loki-debug $(IMAGE_PREFIX)/loki-debug:$(IMAGE_TAG)
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/loki:$(IMAGE_TAG)-debug -f cmd/loki/Dockerfile.debug .
+
+loki-push:
+	$(SUDO) $(PUSH_OCI) $(IMAGE_PREFIX)/loki:$(IMAGE_TAG)
 
+# loki-canary
 loki-canary-image:
-	$(SUDO) docker build -t $(IMAGE_PREFIX)/loki-canary -f cmd/loki-canary/Dockerfile .
-	$(SUDO) docker tag $(IMAGE_PREFIX)/loki-canary $(IMAGE_PREFIX)/loki-canary:$(IMAGE_TAG)
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/loki-canary:$(IMAGE_TAG) -f cmd/loki-canary/Dockerfile .
+loki-canary-push:
+	$(SUDO) $(PUSH_OCI) $(IMAGE_PREFIX)/loki-canary:$(IMAGE_TAG)
 
+# build-image (only amd64)
+build-image: OCI_PLATFORMS=
 build-image:
-	$(SUDO) docker build -t $(IMAGE_PREFIX)/loki-build-image -f loki-build-image/Dockerfile .
-	$(SUDO) docker tag $(IMAGE_PREFIX)/loki-build-image $(IMAGE_PREFIX)/loki-build-image:$(IMAGE_TAG)
-
+	$(SUDO) $(BUILD_OCI) -t $(IMAGE_PREFIX)/loki-build-image:$(IMAGE_TAG) ./loki-build-image
```

Comment:
We're losing the default tag here now that it only tags with the specific version, aren't we?

Reply:
`latest` should only ever occur on master. IMO this should be handled by `./tools/image-tag`, that it returns `latest` on master

Reply:
Okay, resolved this. Has the following behaviour now:

- `master` is tagged as `latest`
- tags are tagged using their version

building branches locally would result in the branch name as the tag

This **removes** the `master-SHA` tags. IMHO this is a bad practice, because it encourages users to use either `master`, which is risky for them, or to lock to old `master-SHA`'s which seemed to work at that time. None of these is any good. 

To prevent this, the `SHA` tags are removed now. The problem of stability should be addressed with more frequent *actually stable versions* instead of historic `master` builds.

Reply:
I'm not sure if we're ready to remove the `master-SHA` tags. We've been relying on them for fine grained control of what we put in production. I'm going to leave this up to @slim-bean

Reply:
Okay, hopefully resolved this in 20f4ce8562bc4a0eaadd8bf12a98b10802383374 now.

I wasn't aware of us relying on `master-SHA` tags in production, I have reverted the above change and added a behaviour close to what has been done so far:

* If a tag is built, it is used for the docker tag as well (tag: `v1.0.0`. image: `grafana/loki:v1.0.0`
* If the `master` branch is built, the image is tagged as `master:SHA`, `master` and `latest`
* If any other branch is built, it is not pushed anyways

Reply:
I'm wondering now if there is any reason to keep the WIP tag (is there any harm in keeping it?) It's a nice way to know the image was built from uncommitted changes, and that it's not really reproducible.  Given that we do push images locally sometimes so we can test them in our clusters, the WIP is a nice reminder to commit your changes

Reply:
There isn't, I can add it back

Reply:
Added back